todo1/src/components/TodoList.js

```diff
@@ -0,0 +1,72 @@
+import React from 'react';
+import { addTodo, toggleTodo} from '../actions';
+import { connect } from 'react-redux';
+
+class TodoListContainer extends React.Component {
+    constructor() {
+      super();
+      this.state = {
+        todoInput: ''
+      };
+    }
+
+    handleChanges = event => {
+        this.setState({ [event.target.name]: event.target.value });
+    };
+
+    handleAddTodo = event => {
+        event.preventDefault();
+        this.props.addTodo(this.state.todoInput);
+        this.setState({
+            todoInput: ''
+        })
+      };
+    
+    handletoggleTodo = index => {
+        this.props.toggleTodo(index);
+    };
+
+    render() {
+        return (
+          <div>
+
+            {this.props.todos.map((todo, index) => (
```

Comment:
It is better to decouple them into different components so it will be reusable if need in anywhere else, I know in this case it won't be needed but it is good practice to get used to.

[Container Components](https://medium.com/@learnreact/container-components-c0e67432e005)